pgmpy/models/MarkovChain.py

```diff
@@ -0,0 +1,299 @@
+    >>> diff_tm = {0: {0: 0.1, 1: 0.5, 2: 0.4}, 1: {0: 0.2, 1: 0.2, 2: 0.6 }, 2: {0: 0.7, 1: 0.15, 2: 0.15}}
+    >>> model.add_transition_model('diff', diff_tm)
+
+    Set a start state
+    >>> model.set_start_state({'intel': 0, 'diff': 2})
+
+    Sample from it
+    >>> model.sample(size=5)
+       intel  diff
+    0      0     2
+    1      1     0
+    2      0     1
+    3      1     0
+    4      0     2
+    """
+    def __init__(self, variables=None, card=None, start_state=None):
+        """
+        Parameters:
+        -----------
+        variables: list
+            a list of variables of the model.
+        card: list
+            a list of cardinalities of the variables.
+        start_state: dict
+            representing the starting states of the variables.
+        """
+        if variables is None:
+            variables = []
+        if card is None:
+            card = []
+        assert isinstance(card, list)
+        assert isinstance(variables, list)
+        self.variables = variables
+        self.cardinalities = {v: c for v, c in zip(variables, card)}
+        self.transition_models = {var: {} for var in variables}
+        if start_state is None or self._check_state(start_state):
+            self.state = start_state
+
+    def set_start_state(self, start_state):
+        """
+        Set the start state of the Markov Chain.
+
+        Parameters:
+        -----------
+        start_state: dict
+            representing the starting states of the variables.
+
+        Examples:
+        ---------
+        >>> from pgmpy.models import MarkovChain as MC
+        >>> model = MC(['a', 'b'], [2, 2])
+        >>> model.set_start_state({'a': 0, 'b': 1})
+        """
+        if start_state is None or self._check_state(start_state):
+            self.state = start_state
+
+    def _check_state(self, state):
+        """
+        Checks if the a dict representation of a states of the variables is valid.
+        """
+        if not isinstance(state, dict):
+            raise ValueError('Start state must be a dict.')
+        if not set(state.keys()) == set(self.transition_models.keys()):
+            raise ValueError('Start state must represent a complete assignment to all variables.')
+        for var, val in state.items():
+            if val >= self.cardinalities[var]:
+                raise ValueError('Assignment {val} to {var} invalid.'.format(val=val, var=var))
+        return True
+
+    def add_variable(self, variable, card=0):
+        """
+        Add a variable to the model.
+
+        Parameters:
+        -----------
+        variable: any hashable python object
+        card: int
+            representing the cardinality of the variable to be added.
+
+        Examples:
+        ---------
+        >>> from pgmpy.models import MarkovChain as MC
+        >>> model = MC()
+        >>> model.add_variable('x', 4)
+        """
+        assert variable not in self.variables
```

Comment:
@pratyakshs I don't think we should be throwing an error even if the variable is already present. Maybe show a warning message showing that the earlier attributes of the variable will be over written.